<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AutoscoperConnect</class>
 <widget class="qMRMLWidget" name="AutoscoperConnect">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>790</width>
    <height>1204</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QPushButton" name="applyButton">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="toolTip">
      <string>Run the algorithm.</string>
     </property>
     <property name="text">
      <string>Launch Autoscoper</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="closeAutoscoper">
     <property name="text">
      <string>Close Autoscoper</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="inputsCollapsibleButton">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Inputs</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Config File</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="ctkPathLineEdit" name="configSelector">
        <property name="label">
         <string/>
        </property>
        <property name="currentPath">
         <string/>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="loadConfig">
        <property name="text">
         <string>Load Config</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="CollapsibleButton">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Tracking File Options</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Volume #</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="selectedVolume">
        <property name="minimum">
         <number>-1</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Type</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QFrame" name="frame">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QRadioButton" name="matrixRadio">
           <property name="text">
            <string>Matrix</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="xyzyprRadio">
           <property name="text">
            <string>xyzypr</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Orientation</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QFrame" name="frame_2">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QRadioButton" name="rowRadio">
           <property name="text">
            <string>Row</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="columnRadio">
           <property name="text">
            <string>Column</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Seperator</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QFrame" name="frame_3">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QRadioButton" name="commaRadio">
           <property name="text">
            <string>Comma</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="spaceRadio">
           <property name="text">
            <string>Space</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="8" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Interpolation</string>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <widget class="QFrame" name="frame_4">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QRadioButton" name="nanRadio">
           <property name="text">
            <string>None (NaN)</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="splineRadio">
           <property name="text">
            <string>Spline</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="10" column="0">
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>Translation Units</string>
        </property>
       </widget>
      </item>
      <item row="10" column="1">
       <widget class="QFrame" name="frame_5">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QRadioButton" name="mmRadio">
           <property name="text">
            <string>mm</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="cmRadio">
           <property name="text">
            <string>cm</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="12" column="0">
       <widget class="QLabel" name="label_14">
        <property name="text">
         <string>Rotation Units</string>
        </property>
       </widget>
      </item>
      <item row="12" column="1">
       <widget class="QFrame" name="frame_6">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QRadioButton" name="degreeRadio">
           <property name="text">
            <string>Degrees</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radRadio">
           <property name="text">
            <string>Radians</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="14" column="0">
       <widget class="QLabel" name="label_15">
        <property name="text">
         <string>Tracking File</string>
        </property>
       </widget>
      </item>
      <item row="14" column="1">
       <widget class="ctkPathLineEdit" name="trackingSelector"/>
      </item>
      <item row="17" column="0" colspan="2">
       <widget class="QPushButton" name="loadTracking">
        <property name="text">
         <string>Load Tracking</string>
        </property>
       </widget>
      </item>
      <item row="16" column="0" colspan="2">
       <widget class="QPushButton" name="saveTracking">
        <property name="text">
         <string>Save Tracking</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="CollapsibleButton_2">
     <property name="text">
      <string>Tracking Optimization</string>
     </property>
     <layout class="QFormLayout" name="formLayout_3">
      <item row="2" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Range</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QFrame" name="frame_7">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Start Frame</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="startFrame"/>
         </item>
         <item>
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>End Frame</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="endFrame"/>
         </item>
         <item>
          <widget class="QCheckBox" name="reverse">
           <property name="text">
            <string>Reverse</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Skip Frame</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="skipFrame">
           <property name="minimum">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>Optization Method</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QFrame" name="frame_8">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <widget class="QRadioButton" name="psoRadio">
           <property name="text">
            <string>Partical Swarm</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="downhillRadio">
           <property name="text">
            <string>Downhill Simplex</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_13">
           <property name="text">
            <string># of Refinements</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="refinements">
           <property name="minimum">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_16">
        <property name="text">
         <string>PSO Parameters</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QFrame" name="frame_9">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_9">
         <item>
          <widget class="QLabel" name="label_17">
           <property name="text">
            <string>Min Limit</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="minLim">
           <property name="decimals">
            <number>1</number>
           </property>
           <property name="minimum">
            <double>-100.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.200000000000000</double>
           </property>
           <property name="value">
            <double>-3.000000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_18">
           <property name="text">
            <string>Max Limit</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="maxLim">
           <property name="decimals">
            <number>1</number>
           </property>
           <property name="minimum">
            <double>-100.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.200000000000000</double>
           </property>
           <property name="value">
            <double>3.000000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_19">
           <property name="text">
            <string>Max Epochs</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="maxEpoch">
           <property name="minimum">
            <number>100</number>
           </property>
           <property name="maximum">
            <number>9999</number>
           </property>
           <property name="value">
            <number>1000</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_20">
           <property name="text">
            <string>Max Stall</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="maxStall">
           <property name="minimum">
            <number>5</number>
           </property>
           <property name="value">
            <number>25</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_21">
        <property name="text">
         <string>Cost Function</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QFrame" name="frame_10">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_10">
         <item>
          <widget class="QRadioButton" name="nccRadio">
           <property name="text">
            <string>Bone Modeles (Normalized Cross-Correlation)</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="sadRadio">
           <property name="text">
            <string>Implant Models (Sum of Abs Difference)</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="6" column="0" colspan="2">
       <widget class="QProgressBar" name="progressBar">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="value">
         <number>0</number>
        </property>
        <property name="invertedAppearance">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="7" column="0" colspan="2">
       <widget class="QPushButton" name="startTrack">
        <property name="text">
         <string>Start Tracking</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_22">
        <property name="text">
         <string>Volume #</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="trackingVolume"/>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qMRMLWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkPathLineEdit</class>
   <extends>QWidget</extends>
   <header>ctkPathLineEdit.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
